src/io/branch/branchster/SplashActivity.java

```diff
@@ -98,7 +62,47 @@ public void run() {
 	    }.start();
 
 	}
-	
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        branch = Branch.getInstance(this.getApplicationContext(), getResources().getString(R.string.bnc_app_key));
+        branch.initSession(new BranchReferralInitListener() {
+            @Override
+            public void onInitFinished(JSONObject referringParams, BranchError error) {
+                Log.i("Branchster", "branch init complete!");
+                try {
+                    MonsterPreferences prefs = MonsterPreferences.getInstance(getApplicationContext());
+                    Intent i;
+                    if (referringParams.has("monster")) {
+
+                        prefs.setMonsterName(referringParams.getString("monster_name"));
+                        prefs.setFaceIndex(referringParams.getInt("face_index"));
+                        prefs.setBodyIndex(referringParams.getInt("body_index"));
+                        prefs.setColorIndex(referringParams.getInt("color_index"));
+
+                        i = new Intent(getApplicationContext(), MonsterViewerActivity.class);
+
+                    } else {
+
+                        if (prefs.getMonsterName() == null) {
```

Comment:
Not sure - I just copied this over from the `onStart` block, but from your feedback let me clean some of this code up